### PR TITLE
MULTIARCH-3668: Update make file to use latest golang

### PR DESCRIPTION
**Pull request by Karthik-K-N**
Updated golang version used for running tests
Updated podman command to remove :Z as its not supported

### CODE REVIEW

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: Makefile [hunk 1]
```diff
@@ -30,7 +30,7 @@ REPO_PATH   ?= github.com/openshift/machine-api-provider-powervs
 LD_FLAGS    ?= -X $(REPO_PATH)/pkg/version.Raw=$(VERSION) -extldflags "-static"
 MUTABLE_TAG ?= latest
 IMAGE        = origin-powervs-machine-controllers
-BUILD_IMAGE ?= registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.19-openshift-4.12
```
[the remark this answers is not in the document]
Comment (Karthik-K-N) [reply Makefile:33]: Yes sure, shall we use this then
```
registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.20-openshift-4.14
```

[the remark this answers is not in the document]
Comment (mkumatag) [reply Makefile:33]: yes, lets use that.

**Review by mkumatag — APPROVED** [review]
/lgtm

again please create a jira item and assign

**Pull request merged**